clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -208,7 +209,8 @@ public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
             return new NetworkClientDelegate.PollResult(heartbeatRequestState.heartbeatIntervalMs, Collections.singletonList(leaveHeartbeat));
         }
 
-        boolean heartbeatNow = membershipManager.shouldHeartbeatNow() && !heartbeatRequestState.requestInFlight();
+//        boolean heartbeatNow = membershipManager.shouldHeartbeatNow() && !heartbeatRequestState.requestInFlight();
+        boolean heartbeatNow = membershipManager.state() == MemberState.LEAVING || membershipManager.shouldHeartbeatNow() && !heartbeatRequestState.requestInFlight();
```

Comment:
Could we add comments to list the cases about `heartbeatNow=true`? the condition check is a bit complicated to me :laughing:

Reply:
This make sense, I will add the comments to clarify this variable.